temporal-sdk/src/main/java/io/temporal/client/WorkflowUpdateTimeoutOrCancelledException.java

```diff
@@ -0,0 +1,36 @@
+/*
+ * Copyright (C) 2022 Temporal Technologies, Inc. All Rights Reserved.
+ *
+ * Copyright (C) 2012-2016 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Modifications copyright (C) 2017 Uber Technologies, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this material except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.temporal.client;
+
+import io.temporal.api.common.v1.WorkflowExecution;
+
+/**
+ * Error that occurs when an update call times out or is cancelled.
+ *
+ * <p>Note, this is not related to any general concept of timing out or cancelling a running update,
+ * this is only related to the client call itself.
+ */
+public class WorkflowUpdateTimeoutOrCancelledException extends WorkflowServiceException {
+  public WorkflowUpdateTimeoutOrCancelledException(
```

Comment:
May be worth docs here stating this constructor is for internal use only and that the parameters may not be stable, but maybe not if we never expect to change or we can overload

Reply:
I think it is fine as is, we do the same thing for all the other children of `WorkflowServiceException`. A user may construct it for mocking/testsing

Reply:
Yeah, my only concern is that we can't ever change it, only overload it. But no big deal.